File: app/src/main/java/org/tvheadend/tvhclient/data/entity/TagAndChannel.java
package org.tvheadend.tvhclient.data.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;

@Entity(tableName = "tags_and_channels", primaryKeys = {"tag_id", "channel_id", "connection_id"})
public class TagAndChannel {

    @ColumnInfo(name = "tag_id")
    private int tagId;
    @ColumnInfo(name = "channel_id")
    private int channelId;
    @ColumnInfo(name = "connection_id")
    private int connectionId;

    public int getChannelId() {
        return channelId;
    }

    public void setChannelId(int channelId) {
        this.channelId = channelId;
    }

    public int getTagId() {
        return tagId;
    }

    public void setTagId(int tagId) {
        this.tagId = tagId;
    }

    public int getConnectionId() {
        return connectionId;
    }

    public void setConnectionId(int connectionId) {
        this.connectionId = connectionId;
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/domain/entity/TagAndChannel.kt
package org.tvheadend.tvhclient.domain.entity

import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity(tableName = "tags_and_channels", primaryKeys = ["tag_id", "channel_id", "connection_id"])
data class TagAndChannel(

        @ColumnInfo(name = "tag_id")
        var tagId: Int = 0,
        @ColumnInfo(name = "channel_id")
        var channelId: Int = 0,
        @ColumnInfo(name = "connection_id")
        var connectionId: Int = 0
)
